### ISSUE TITLE
filtering works only for current page

### ISSUE THREAD

**Issue by g-sky** [DanWahlin/Angular-ASPNET-Core-CustomersService#2]
Hi Dan!
Thanks for putting this pluralsight course together, very helpful.
After we add pagination, filterbox is going to filter only customers on current page. And if 'Jack' is on third page and we currently on the first page, no customers are going to be displayed, right?
Do you have code somewhere where 'Jack'(s) from all pages are going to be displayed.
Do you recommend any third party grids for Angular2+?

**Comment by DanWahlin** [comment]
Thanks - glad to hear you enjoyed the course!

Filtering only occurs on the current page that's loaded. If you wanted to filter all customers you'd have to extend the existing RESTful service to add that functionality (no code currently exists in the project for that functionality). It'd be something that could potentially be done with the paging API by adding another parameter that would be used for the filter. Or, you could create a separate filtering API but then you'd have to work paging into it as well (which is why it's probably easiest to extend the existing paging endpoint).

As far as grids, check out AG Grid (https://ag-grid.com/best-angular-2-data-grid/#gsc.tab=0). It started out originally with AngularJS (and can render a lot of data very fast) and they have an Angular 2+ version as well now (in addition to other libraries/frameworks). There are a lot of other ones out there, but that's one of the more popular ones I'm seeing a lot of enterprises that we work with using. 

Happy New Year!

**Comment by g-sky** [comment]
Hi Dan! Appreciate your quick response. 
Thanks for showing water! I knew how to drink it :) 
I made it working for me using your code, just a couple tweaks:  
OnInit I got AllCustomers (not only the first page). I can afford it – not too much data. This way I have it available for filtering.  
Grid source is still filteredCustomers.  
I got filteredCustomers from the same Response I got AllCustomers, just sliced first 10 from array of AllCustomers. 
This same slice (first 10) is assigned to filterCustomers again if filterText isNullOrEmpty. 

Thanks for your AG Grid recommendation. 

Happy New Year! 
Sincerely, 
Gene.
